pennylane/devices/qubit/measure.py

```diff
@@ -86,7 +88,10 @@ def csr_dot_products(
         TensorLike: the result of the measurement
     """
     total_wires = len(state.shape) - is_state_batched
-    Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
+    if not is_state_batched and is_pauli_sentence(measurementprocess.obs):
+        ps = pauli_sentence(measurementprocess.obs)
+    else:
+        Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
```

Comment:
"sometimes defined" variables always give me a bit of pause.  If we took one logical branch, `ps` would be a valid variable, but we took the other branch, then it wouldn't be defined.

Instead, we could just calculate the quantity inside the logical branch where we squeeze it. We would then be calculating `Hmat` separately on each branch, but I think that's an acceptable level of duplication.

Reply:
Agreed.